### DISCUSSION TITLE
dotnet build's parameter NetCoreBuild is not defined

### DISCUSSION THREAD

**Discussion by kev1010** [microsoft/DacFx#186] [Q&A] [answered]
[Read me](https://github.com/microsoft/DacFx/tree/main/src/Microsoft.Build.Sql) documents the build process with:
`dotnet build /p:NetCoreBuild=true`

However,  [dotnet build](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build) does not define the `NetCoreBuild` parameter. Where can I find the documentation for this parameter?

**Comment by ErikEJ** [comment]
You do not need it with the latest SDK version

**Reply by kev1010** [reply]
This is a question about the documentation in the repository.

**Comment by stijnherreman** [comment, marked as answer]
`/p` parameters in general are documented at https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build#msbuild

`NetCoreBuild` is not a generic parameter, it is specific to this SDK. You can see how it is used in https://github.com/microsoft/DacFx/blob/61888b2c75bfdcfd22ab69a0f25b7fdbcde93b6e/src/Microsoft.Build.Sql/sdk/Sdk.targets#L10-L12

Instead of passing the parameter via `dotnet build`, you can also configure it in your `sqlproj` instead:
```xml
  <PropertyGroup>
    <NetCoreBuild>true</NetCoreBuild>
  </PropertyGroup>
```

However, as ErikEJ says, this is no longer necessary with the latest SDK version `0.1.7-preview`.

**Reply by kev1010** [reply]
Thank you @stijnherreman  and @ErikEJ  for your responses. It seems this parameter was used for integration with SSDT based on this comment in the Readme file:
``` Note: Building with .NET Core targets is currently unsupported in SSDT. Please make sure the NetCoreBuild property is false or not defined when building in SSDT. ```
From the comments in our discussion, it seems like this limitation is no longer a concern. Therefore, shouldn't the parameter be removed altogether to clean up the code?

I have referenced this thread in [PR 191](https://github.com/microsoft/DacFx/pull/191) where the author is updating other documentation to version 0.1.7-preview.

**Reply by zijchen** [reply]
The `NetCoreBuild` property is still needed for the build process to load the correct assemblies. When you build with `dotnet build`, `NetCoreBuild` is true and it will use the assemblies packaged in this SDK. When you build with SSDT, `NetCoreBuild` is false and it will use the assemblies installed by SSDT. The latest SDK version just handles setting the property for you.

We hope to clean this up so SSDT also uses the assemblies from the SDK.

**Comment by kev1010** [comment]
Why is it so complicated to make them both use the same build? Wouldn't it
just be changing it to use the same? In any case, it would be helpful for
it to be documented in the help file so people would know what and why the
builds are doing what they do.
